### ISSUE TITLE
_test_tinshift_assert

### ISSUE THREAD

**Issue by defei-coder** [open-mmlab/mmcv#2518]
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the FAQ documentation but cannot get the expected help.

Hi！
 I found two unreasonable code in [_test_tinshift_assert](https://github.com/open-mmlab/mmcv/blob/master/tests/test_ops/test_tin_shift.py#L182). First, the numbers of input and shift is mismatch, so the torch.rand(1, 3, 4, 2) of inputs will not be tested. Second, `shift = shift.to(device).type(dtype)`, shift should be int, or it will cause runtimeerror, although it will not cause an error here.

**Comment by HAOCHENYE** [comment]
Thanks for your feedback :smile: 

Firstly, this line here is actually unreasonable, maybe we should add one item to `shift` or delete this line.
https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L191

As for `shift = shift.to(device).type(dtype)`. dtype is controlled by this function

https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L214

Therefore the type will note be `int` here.

**Comment by defei-coder** [comment]
> Thanks for your feedback smile
> 
> Firstly, this line here is actually unreasonable, maybe we should add one item to `shift` or delete this line.
> 
> https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L191
> 
> As for `shift = shift.to(device).type(dtype)`. dtype is controlled by this function
> 
> https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L214
> 
> Therefore the type will note be `int` here.

About the shift tensor, I know the dtype, but I am confused by your reply. I mean current implement is a bug [](https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L193) the right implement is `shift = shift = shift.to(device).int().` So do you agree with me?

**Comment by HAOCHENYE** [comment]
You're right, other unit tests in this file convert the type of shift by `shift = torch.tensor(np_shift, device=device).int()`

here
https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L151

and 

https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L172.

Besides, the initial value of `shifts = [torch.rand(2, 3), torch.rand(2, 5)]` is also weird, it should be a int array just like https://github.com/open-mmlab/mmcv/blob/2810718a99045bb762d24f465a69b7add34f3123/tests/test_ops/test_tin_shift.py#L53.

Thanks for your feedback! Would you mind posting a PR to fix this?
